.github/workflows/tests.yaml

```diff
@@ -261,8 +258,38 @@ jobs:
           # build, so simply inject an explicit no-op test target.
           echo "//scripts:no_op_test" >> $TARGETS_FILE
 
-          bazelisk test -c ${{ matrix.build_mode }} \
-            --target_pattern_file=$TARGETS_FILE
+          for i in {1..10}; do
+            bazel_exit=0
+            bazelisk test -c ${{ matrix.build_mode }} \
+              --target_pattern_file=$TARGETS_FILE || bazel_exit=$?
```

Comment:
It seems like we should decrease the `--jobs` after a failure, to adapt to an overloaded environment.

Reply:
I was really hesitant to add even more logic here, and I'm not sure how important this will be compared to just retries... All the downloads that finish before the timeout or IO failure will be cached and so each retry should actually have significantly less work to do (and so end up with less load).

Anyways, I see a simpler way to add this now, so let's try it.